src/Compilers/CSharp/Portable/Compilation/MemberSemanticModel.cs

```diff
@@ -1914,13 +1917,15 @@ private static Binder GetLambdaEnclosingBinder(int position, CSharpSyntaxNode st
         /// </summary>
         protected void EnsureNullabilityAnalysisPerformedIfNecessary()
         {
-            // If we're in DEBUG mode, always enable the analysis, but throw away the results
-#if !DEBUG
+            // If we're in DEBUG mode, enable the analysis unless explicitly disabled, but throw away the results
+#if DEBUG
+            if (!Compilation.ShouldRunNullableAnalysisAndIgnoreResults)
```

Comment:
>ShouldRunNullableAnalysisAndIgnoreResults [](start = 29, length = 41)

The name of this property (the "AndIgnoreResults" part) feels confusing in context of this ```if```. Do we return even though we don't want to ignore the result? I think we either need to come up with a "better" name, or slightly adjust the code. For example:
```
            if (!Compilation.NullableSemanticAnalysisEnabled
#if DEBUG
           && !Compilation.ShouldRunNullableAnalysisAndIgnoreResults
#endif
         )
            {
                return;
            }
``` 

Even though combining both conditions in DEBUG might feel redundant if implementation of properties is "inlined", I think the intent is expressed much clearer and is easier to understand. The logic will be more robust to future changes in implementation of these properties. Also, this matches the logic we have below around the same conditions. In fact, that logic can be simplified. We can remove the added ```if (Compilation.ShouldRunNullableAnalysisAndIgnoreResults)``` and assert that the condition is true instead.